src/utils.py

```diff
@@ -11,6 +12,36 @@
 
 import glob, os
 
+from zipfile import ZipFile
+import re
+
+# Zip the files from given directory that matches the filter
+def zip_dir_filter(dir_name, zip_Name, files_filter=None):
+    '''
+    Create zip file of folder. Folder will be recursively searched.
+    Allows to specify a list of regular expression to included file-names.
+    '''
+
+    with ZipFile(zip_Name, 'w') as zipObj:
+        # Iterate over all the files in directory
+        for foldername, subfolders, filenames in os.walk(dir_name):
+            for filename in filenames:
+
+                file_include = True
+
+                if files_filter:
+                    if [ True for file_filter in files_filter if re.search(file_filter, filename) ] == []:
+                        file_include = False
+
+                if file_include:
+                    filePath = os.path.join(foldername, filename)
+                    arcname = filePath.replace(dir_name,'')
+                    zipObj.write(filePath,arcname)
+
+    # close the Zip File
+    zipObj.close()
```

Comment:
Here also, you don't need to close it since it's already closed when you exit the `with` block.